tests/genai/judges/optimizers/test_dspy_base.py

```diff
@@ -320,3 +320,64 @@ def mock_dspy_optimize(program, examples, metric_fn):
         # Verify suppression was active during the DSPy optimization call
         assert suppression_state_during_call["set_verbose"] is False
         assert suppression_state_during_call["suppress_debug_info"] is True
+
+
+@pytest.mark.parametrize(
+    ("model", "expected_type"),
+    [
+        ("databricks", "AgentEvalLM"),
+        ("openai:/gpt-4", "dspy.LM"),
+        ("anthropic:/claude-3", "dspy.LM"),
+    ],
+)
+def test_construct_dspy_lm_utility_method(model, expected_type):
+    """Test the _construct_dspy_lm utility method with different model types."""
+    from mlflow.genai.judges.optimizers.dspy import _construct_dspy_lm
```

Comment:
Not a big fan of testing private methods. Let's make this a public method in dpsy_utils and test it there.